opentracing-spring-messaging/src/main/java/io/opentracing/contrib/spring/integration/messaging/OpenTracingChannelInterceptor.java

```diff
@@ -66,32 +67,42 @@ public OpenTracingChannelInterceptor(Tracer tracer) {
       spanBuilder.asChildOf(extractedContext);
     }
 
-    Span span = spanBuilder.startActive(true).span();
+    Span span = spanBuilder.start();
+    tracer.activateSpan(span);
```

Comment:
This will create leaks. The scope should be closed

Reply:
If you need to create a scope then create it in a single line when you are starting the span

Reply:
How do you do that with the new API? I don't see any way. In my next commit I have taken care of propagating the scope to close it at the end of the message handling.